src/main/java/org/mockbukkit/mockbukkit/block/data/CandleDataMock.java

```diff
@@ -0,0 +1,69 @@
+package org.mockbukkit.mockbukkit.block.data;
+
+import org.bukkit.Material;
+import org.bukkit.block.data.type.Candle;
+import org.jetbrains.annotations.NotNull;
+
+public class CandleDataMock extends BlockDataMock implements Candle
+{
+
+	/**
+	 * Constructs a new {@link CandleDataMock} for the provided {@link Material}.
+	 *
+	 * @param material The material this data is for.
+	 */
+	protected CandleDataMock(@NotNull Material material)
+	{
+		super(material);
+	}
+
+
+	@Override
+	public int getCandles()
+	{
+		return this.get(BlockDataKey.CANDLES);
+	}
+
+	@Override
+	public void setCandles(int candles)
+	{
+		this.set(BlockDataKey.CANDLES, candles);
```

Comment:
This value should also be truncated between the maximum and minimum allowed candles